### ISSUE TITLE
Add interface for class-based schemas

### ISSUE THREAD

**Issue by fisxoj** [fisxoj/sanity-clause#2]
This got left out of the big rework, but it would be good to have a class based schema interface on top of the plist version that's curretly available.

The interface should resemble:

```lisp
(defclass my-app-config ()
  ((port :field (integer))
   (username :field (string :validator (not-empty)))
  (:metaclass sanity-clause:schema-class))
```

The field should be class-allocated per slot.  The `:field` initarg should hand things off to `make-field`, probably!

**Comment by fisxoj** [comment]
Closed by #5

**Comment by vindarel** [comment]
Nicely done.
